modules/planning/tasks/deciders/speed_bounds_decider/speed_bounds_decider.cc

```diff
@@ -159,6 +164,51 @@ Status SpeedBoundsDecider::Process(
   return Status::OK();
 }
 
+void SpeedBoundsDecider::CheckLaneChangeUrgency(Frame *const frame) {
+  // Decide if more than one reference lines exist
+  if (frame->reference_line_info().size() > 1) {
+    for (auto& reference_line_info : *frame->mutable_reference_line_info()) {
+      // If current lane is target lane
+      if (reference_line_info.IsChangeLanePath()) {
+        is_clear_to_change_lane_ = \
+            ChangeLaneDecider::IsClearToChangeLane(&reference_line_info);
+      } else {
+        // If lane change is blocked
+        if (!is_clear_to_change_lane_) {
+          // Get the end point of current routing
+          const auto& route_end_waypoint = \
+          reference_line_info.Lanes().RouteEndWaypoint();
+          if (route_end_waypoint.lane) {
+            auto point = \
+            route_end_waypoint.lane->GetSmoothPoint(route_end_waypoint.s);
+            auto* reference_line = reference_line_info.mutable_reference_line();
+            common::SLPoint sl_point;
+            // Project the end point to current reference lane
+            if (reference_line->XYToSL({point.x(), point.y()}, &sl_point) && \
+                  reference_line->IsOnLane(sl_point)) {
+              // Distance to the end of current routing
+              double distance_to_passage_end = sl_point.s() - \
+                  reference_line_info.AdcSlBoundary().end_s();
+              // If adc near the end of routing, stop and wait
+              if (distance_to_passage_end < \
+                  speed_bounds_config_.approach_distance_for_lane_change()) {
+                // Set a stop fence and wait for chances to change lane
+                const std::string stop_wall_id = "lane_change_stop";
+                std::vector<std::string> wait_for_obstacles;
+                DeciderRuleBasedStop::BuildStopDecision(
+                       stop_wall_id, sl_point.s(),
+                       speed_bounds_config_.urgent_distance_for_lane_change(),
+                       StopReasonCode::STOP_REASON_LANE_CHANGE_URGENCY,
+                       wait_for_obstacles, frame, &reference_line_info);
```

Comment:
Try to reduce indents. Some practices:

1. You can check the negative logic and call `continue`.
```
if (is_clear_to_change_lane_) {
  continue;
}
// Other logic.
```

2. You can wrap inside logic into function to make it more readable:
```
if (!is_clear_to_change_lane_) {
  DealWithIt();
}
```

Reply:
Solved. Thanks!